drivers/video/arducam_mega.c

```diff
@@ -0,0 +1,1064 @@
+/**
+ * The MIT License (MIT)
+ *
+ * Copyright 2021 Arducam Technology co., Ltd. All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE
+ *
+ */
+
+#define DT_DRV_COMPAT arducam_mega
+
+#include <zephyr/drivers/video/arducam_mega.h>
```

Comment:
Why do you need an header file? as far as I see everything can be in the C file.

Reply:
@loicpoulain 
Thank you for your detail check.
Due to our arducam mega driver need some specific controls and other parameters and we need provide them to user.
We have not see them in the video.h file. So we add a arducam_mega.h file. If not, users can't control our mega's full features.

Reply:
Ok, but look like some of them, like exposure, brightness, etc, are quite common for camera, so it would be worth to add new generic CId for them, instead of having custom ones for each vendor. For sure that new control values should be in known unit like `us` or `%`..

Reply:
@loicpoulain 
Hello,  Thank you for your detail check. Let me understand your means.
For some common camera controls, it is not possible to define custom own CID, Just use the commn CID in the video.h.
For some specific control, we can define new CID to video.h. Right?
Sorry I don't get it about  "For sure that new control values should be in known device-agnostic unit like us or %..".
How to modify it.

Reply:
@loicpoulain 
Hello, thanks for your fantastic sugguestions. In fact， in the video-controls.h file there are exist some common  camera controls, and we have use it，we can add some other common controls（such as exposure mode） to video-controls.h and send you a  new PR.
But，we have some specific control just used for arducam mega. so we can't make sure all controls are common.
Do you think so?

Reply:
Yes that's fine, you can also have private controls. But it makes sense to make the generic ones common.

Reply:
@loicpoulain 
OK, I have modified it.